consensus/src/units/store.rs

```diff
@@ -28,6 +79,46 @@ impl<'a, H: Hasher, D: Data, KB: KeyBox> UnitStore<'a, H, D, KB> {
         }
     }
 
+    pub fn get_status(&self) -> UnitStoreStatus {
+        fn first_missing(v: Vec<Round>) -> Round {
+            let mut w = vec![false; v.len()];
+            for r in &v {
+                if let Some(f) = w.get_mut(*r as usize) {
+                    *f = true;
+                }
+            }
+            match w.into_iter().position(|x| !x) {
+                Some(r) => r as u16,
+                None => v.len() as u16,
+            }
+        }
+        let gm = self
+            .by_coord
+            .keys()
+            .map(|c| (c.creator.0, c.round))
+            .into_grouping_map();
+        let top_row: HashMap<usize, Round> = gm.clone().max();
+        let mut first_missing_rounds: Vec<(usize, Round)> = gm
+            .collect::<Vec<_>>()
+            .into_iter()
+            .map(|(id, rounds)| (id, first_missing(rounds)))
+            .filter(|(id, round)| match top_row.get(id) {
+                Some(top_round) => round < top_round,
+                None => false,
+            })
+            .collect();
+        let mut top_row: Vec<(usize, Round)> = top_row.into_iter().collect();
```

Comment:
Ah, I think I see, you want this sorted...

It might be better to keep these as `HashMap`s and print them out more explicitly in the `Display` implementation, e.g. by using something like `(1..node_count).for_each(|node_id| top_row[node_id]...)`.

Reply:
added `Display` to `NodeMap`